src/utils/cpuProfileParsing.ts

```diff
@@ -0,0 +1,119 @@
+import _ from 'lodash';
+import {
+  HermesSample,
+  HermesStackFrame,
+} from '../types/hermesProfileInterfaces';
+import { CallGraph, Node } from './callGraph';
+import { SharedEventProperties } from 'types/chromeEventInterfaces';
+
+/* Converts our data into an intermediate form used for processing
+ * For Example: samples => [{
+ *    "cpu": "-1",
+ *    "name": "",
+ *    "ts": "142867379",
+ *    "pid": 6052,
+ *    "tid": "6105",
+ *    "weight": "1",
+ *    "sf": 16
+ *}]. This function converts it to '6052:6105': [{
+ *    "cpu": "-1",
+ *   "name": "",
+ *    "ts": "142867379",
+ *    "pid": 6052,
+ *    "tid": "6105",
+ *    "weight": "1",
+ *    "sf": 16
+ *}]
+ */
+const getPidTid = (data: HermesSample[]) => {
+  const pidTid: { [key in string]: HermesSample[] } = {};
+  data.forEach((sample: HermesSample) => {
+    const property = `${sample.pid}:${sample.tid}`;
+    if (!(property in pidTid)) {
+      pidTid[property] = [];
+    }
+    pidTid[property].push(sample);
+  });
+  return pidTid;
+};
+
+/* 
+Obtain List of All processes in the Hermes Profile
+ */
+const getProcesses = (data: HermesSample[]) => {
+  return data
+    .map((sample: HermesSample) => sample.pid)
+    .filter((value, index, self) => self.indexOf(value) === index);
+};
+
+export const processSample = (
+  samples: HermesSample[],
+  stackFrames: { [key in string]: HermesStackFrame }
+) => {
+  return samples.map((sample: HermesSample) => {
+    const stackFrameId: string = sample.sf.toString();
+    sample.stackFrameData = { ...stackFrames[stackFrameId] };
+    sample.name = sample.stackFrameData.name;
+    return sample;
+  });
+};
+
+const findRootNode = (
```

Comment:
This function finds the root node for every process. This is primarily used to initialise our process graph, and find the root node for a particular process.